src/transformers/models/modernbert/modeling_modernbert.py

```diff
@@ -667,6 +670,13 @@ def _maybe_set_compile(self):
                 )
             self.config.reference_compile = False
 
+        if self.device.type == "cpu":
+            logger.warning_once(
```

Comment:
I think we should wrap this in `if self.config.reference_compile:` just like above so users don't get warned when they never manually specified anything related to `reference_compile`.

Unless you'd rather always give this warning so people aren't confused why there's a small precision-esque difference between GPU setups?